targets/TARGET_STM/hal_tick_overrides.c

```diff
@@ -47,26 +46,28 @@ HAL_StatusTypeDef HAL_InitTick(uint32_t TickPriority)
 
 uint32_t HAL_GetTick()
 {
-#if TIM_MST_BIT_WIDTH == 16
-    uint32_t new_time;
-    if (mbed_sdk_inited) {
-        // Apply the latest time recorded just before the sdk is inited
-        new_time = ticker_read_us(get_us_ticker_data()) + prev_time;
-        prev_time = 0; // Use this time only once
-        return (new_time / 1000);
+    uint32_t new_time = us_ticker_read();
+    uint32_t elapsed_ticks = 0;
+    uint32_t elapsed_time = (((new_time - prev_time) & US_TICKER_MASK) + prev_tick_remainder);
+    prev_time = new_time;
+    if (elapsed_time < 1000) { // elapsed time less than 1ms
```

Comment:
These two special cases seem to me to be pointless code bloat.  I don't see how they're going to be noticeably faster than just going into the while loop. We generally want to optimise for space, except where we absolutely have to.

They'd make sense if you didn't have the while loop and the alternative was the standard C library division.